### PR TITLE
fix(ui_auth): make light text on _EmailVerificationBadge legible

### LINKED ISSUE #18: 🐛 [Firebase_UI_Auth] Verification message text not visible in Profile Screen due to wrong color
## Bug report

**Describe the bug**
In the Profile Screen, the email verification message is not visible in Dark Mode.
My app color theme is yellow, and the background of the message is also very light colored as is the text, and it is not clearly visible.

### Steps to reproduce

1. Create a flutter project
2. Set the flutter theme `useMaterial3: true`, and colorScheme to `ColorScheme.fromSeed(brightness: Brightness.dark, seedColor: Colors.yellow)`
3. Use Firebase UI Auth to Log In and then display the Profile Screen
4. See the rendering issue

### Expected behavior

The text adapts to be always visible in any condition.

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.22621.1702], locale es-ES)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.8.2)
[√] Android Studio (version 2022.1)
[√] VS Code (version 1.78.2)
[√] Connected device (5 available)
[√] Network resources

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
dependencies:
- animated_text_kit 4.2.2 [flutter characters]
- cloud_firestore 4.7.0 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- connectivity_plus 4.0.0 [flutter flutter_web_plugins connectivity_plus_platform_interface js meta nm]
- dynamic_color 1.6.5 [flutter flutter_test material_color_utilities]
- firebase_analytics 10.4.0 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_app_check 0.1.4 [firebase_app_check_platform_interface firebase_app_check_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 4.6.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.12.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 3.3.0 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_performance 0.9.2 [firebase_core firebase_core_platform_interface firebase_performance_platform_interface firebase_performance_web flutter]
- firebase_remote_config 4.2.0 [firebase_core firebase_core_platform_interface firebase_remote_config_platform_interface firebase_remote_config_web flutter]
- firebase_storage 11.2.0 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- firebase_ui_auth 1.4.0 [email_validator firebase_auth firebase_core firebase_dynamic_links firebase_ui_localizations firebase_ui_oauth firebase_ui_shared flutter flutter_localizations flutter_svg]
- firebase_ui_oauth_apple 1.2.0 [firebase_auth firebase_ui_oauth flutter]
- firebase_ui_oauth_google 1.2.0 [firebase_auth firebase_ui_oauth flutter google_sign_in]
- flutter 0.0.0 [characters collection js material_color_utilities meta vector_math sky_engine]
- flutter_animate 4.1.1+1 [flutter]
- flutter_localizations 0.0.0 [flutter intl characters clock collection js material_color_utilities meta path vector_math]
- go_router 7.0.1 [collection flutter flutter_web_plugins logging meta]
- google_fonts 4.0.4 [flutter http path_provider crypto]
- google_sign_in 6.1.0 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- provider 6.0.5 [collection flutter nested]
- shared_preferences 2.1.1 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- url_launcher 6.1.11 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]

dev dependencies:
- dependency_validator 3.2.2 [args build_config checked_yaml glob io json_annotation logging package_config path pub_semver pubspec_parse yaml]
- flutter_lints 2.0.1 [lints]
- flutter_native_splash 2.2.19 [args flutter flutter_web_plugins js html image meta path universal_io xml yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters collection js matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

transitive dependencies:
- _flutterfire_internals 1.3.0 [collection firebase_core firebase_core_platform_interface flutter meta]
- archive 3.3.7 [crypto path pointycastle]
- args 2.4.1
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- build_config 1.1.1 [checked_yaml json_annotation path pubspec_parse yaml]
- characters 1.3.0
- checked_yaml 2.0.3 [json_annotation source_span yaml]
- clock 1.1.1
- cloud_firestore_platform_interface 5.14.0 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.5.0 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.17.1
- connectivity_plus_platform_interface 1.2.4 [flutter meta plugin_platform_interface]
- convert 3.1.1 [typed_data]
- crypto 3.0.3 [typed_data]
- csslib 0.17.2 [source_span]
- dbus 0.7.8 [args ffi meta xml]
- desktop_webview_auth 0.0.12 [crypto flutter http flutter_web_plugins plugin_platform_interface]
- email_validator 2.1.17
- fake_async 1.3.1 [clock collection]
- ffi 2.0.2
- file 6.1.4 [meta path]
- firebase_analytics_platform_interface 3.6.0 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.5.4 [_flutterfire_internals firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_app_check_platform_interface 0.0.8 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_app_check_web 0.0.10 [_flutterfire_internals firebase_app_check_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_auth_platform_interface 6.15.0 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.5.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]
- firebase_core_platform_interface 4.8.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.5.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.6.0 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_dynamic_links 5.3.0 [firebase_core firebase_core_platform_interface firebase_dynamic_links_platform_interface flutter meta plugin_platform_interface]
- firebase_dynamic_links_platform_interface 0.2.6 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_performance_platform_interface 0.1.4 [_flutterfire_internals firebase_core flutter plugin_platform_interface]
- firebase_performance_web 0.1.4 [_flutterfire_internals firebase_core firebase_core_web firebase_performance_platform_interface flutter flutter_web_plugins js]
- firebase_remote_config_platform_interface 1.4.0 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_remote_config_web 1.4.0 [firebase_core firebase_core_web firebase_remote_config_platform_interface flutter flutter_web_plugins js]
- firebase_storage_platform_interface 4.4.0 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 3.6.0 [_flutterfire_internals async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- firebase_ui_localizations 1.5.0 [flutter flutter_localizations path]
- firebase_ui_oauth 1.4.0 [desktop_webview_auth firebase_auth firebase_ui_auth firebase_ui_shared flutter_svg flutter]
- firebase_ui_shared 1.3.0 [flutter]
- flutter_svg 1.1.6 [flutter meta path_drawing vector_math xml]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- glob 2.1.1 [async collection file path string_scanner]
- google_identity_services_web 0.2.1 [js meta]
- google_sign_in_android 6.1.14 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 5.6.2 [flutter google_sign_in_platform_interface]
- google_sign_in_platform_interface 2.4.0 [flutter plugin_platform_interface quiver]
- google_sign_in_web 0.12.0 [flutter flutter_web_plugins google_identity_services_web google_sign_in_platform_interface http js]
- html 0.15.3 [csslib source_span]
- http 0.13.6 [async http_parser meta]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 4.0.17 [archive meta xml]
- intl 0.18.0 [clock meta path]
- io 1.0.4 [meta path string_scanner]
- js 0.6.7 [meta]
- json_annotation 4.8.1 [meta]
- lints 2.1.0
- logging 1.1.1
- matcher 0.12.15 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.2.0
- meta 1.9.1
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- package_config 2.1.0 [path]
- path 1.8.3
- path_drawing 1.0.1 [vector_math meta path_parsing flutter]
- path_parsing 1.0.1 [vector_math meta]
- path_provider 2.0.15 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.27 [flutter path_provider_platform_interface]
- path_provider_foundation 2.2.3 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.10 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.6 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.6 [ffi flutter path path_provider_platform_interface win32]
- petitparser 5.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.4 [meta]
- pointycastle 3.7.3 [collection convert js]
- process 4.2.4 [file path platform]
- pub_semver 2.1.4 [collection meta]
- pubspec_parse 1.2.3 [checked_yaml collection json_annotation pub_semver yaml]
- quiver 3.2.1 [matcher]
- shared_preferences_android 2.1.4 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.2.2 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.2.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.2.0 [flutter plugin_platform_interface]
- shared_preferences_web 2.1.0 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.2.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- sky_engine 0.0.99
- source_span 1.9.1 [collection path term_glyph]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.5.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.2 [collection]
- universal_io 2.2.0 [collection meta typed_data]
- url_launcher_android 6.0.33 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.1.4 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.5 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.5 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.2 [flutter plugin_platform_interface]
- url_launcher_web 2.0.16 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.6 [flutter url_launcher_platform_interface]
- vector_math 2.1.4
- win32 4.1.4 [ffi]
- xdg_directories 1.0.0 [meta path process]
- xml 6.3.0 [collection meta petitparser]
- yaml 3.1.2 [collection source_span string_scanner]
```

</details>

---

### PR DESCRIPTION
**Pull request by qwezey**
## Description

- Reduced the opacity of the badge's background to make light text in a dark theme more legible
- Dark text in a light theme is still legible
- The reduced opacity also makes the color more appealing

## Related Issues

fixes #18 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for _all_ changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] All unit tests pass (`melos run test:unit:all` doesn't fail).
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is _not_ a breaking change.

[issue database]: https://github.com/firebase/FirebaseUI-Flutter/issues
[Contributor Guide]: https://github.com/firebase/FirebaseUI-Flutter/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by EArminjon** [comment]
I think it could be better to expose this color through a dedicated parameter. In fact, it could be good to override every widget UI of this package to custom them as close as client app current design.
It could be really fantastic to be able to have the hand on every things :/ !

**Comment by qwezey** [comment]
@EArminjon For this PR, do you think I should allow the client to override the whole `decoration` property of the `Container` or only the `color` property of the `BoxDecoration`?

**Comment by lesnitsky** [comment]
> it could be good to have the possibility to override every UI widget of this package to custom them as close as client app current design.

Styling API is in place and contributions on making anything configurable are welcome.

In general, here's how styling should be taken care of:

- extend [FirebaseUIStyle](https://github.com/firebase/FirebaseUI-Flutter/blob/7fb42d3704eeae26d09b0f8fa3f1012a9f0ac887/packages/firebase_ui_auth/lib/src/styling/style.dart#L11)
- override `applyToMaterialTheme` or `applyToCupertinoTheme` if necessary (see [this example](https://github.com/firebase/FirebaseUI-Flutter/blob/7fb42d3704eeae26d09b0f8fa3f1012a9f0ac887/packages/firebase_ui_auth/lib/src/widgets/email_form.dart#L57))
- define a property on a relevant widget (like [here](https://github.com/firebase/FirebaseUI-Flutter/blob/7fb42d3704eeae26d09b0f8fa3f1012a9f0ac887/packages/firebase_ui_auth/lib/src/widgets/email_form.dart#L132))
- define `Set<FirebaseUIStyle> styles` on a screen level (like [here](https://github.com/firebase/FirebaseUI-Flutter/blob/7fb42d3704eeae26d09b0f8fa3f1012a9f0ac887/packages/firebase_ui_auth/lib/src/screens/internal/login_screen.dart#L52))
- use `FirebaseUIStyle.ofType<T extends FirebaseUIStyle> (like [here](https://github.com/firebase/FirebaseUI-Flutter/blob/7fb42d3704eeae26d09b0f8fa3f1012a9f0ac887/packages/firebase_ui_auth/lib/src/widgets/email_form.dart#L318)) to get a relevant style object from build context

In this specific issue, we could create an `EmailVerificationBadgeStyle` object and have a configurable `backgroundColor` defined there.

@qwezey would you be willing to update your PR accordingly?

File: packages/firebase_ui_auth/lib/src/screens/profile_screen.dart [hunk 1]
```diff
@@ -387,7 +387,7 @@ class _EmailVerificationBadgeState extends State<_EmailVerificationBadge> {
         children: [
           Container(
             decoration: BoxDecoration(
-              color: Colors.yellow,
+              color: Colors.yellow.withOpacity(0.5),
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/screens/profile_screen.dart:390]: please make this configurable as mention [here](https://github.com/firebase/FirebaseUI-Flutter/pull/196#issuecomment-1832483440).

**Comment by russellwheatley** [comment]
Hey @qwezey - will you be following up on this PR?

**Comment by cmjordan42** [comment]
Can we proceed with merging this PR? It is a bug fix but requests to incorporate non-critical enhancements seem to be stalling it. We are at a 0 currently and this change seems like it would get us to a 1, so it seems prudent to keep any additional changes to get to a 2 separate from this.

**Review by russellwheatley — APPROVED** [review]

**Comment by cmjordan42** [comment]
Many thanks, @russellwheatley

**Pull request merged**